dev-docs/RFCs/v4.0.0/terms_for_concepts_and_datatypes.md

```diff
@@ -0,0 +1,96 @@
+
+
+## Use of "Topological"
+
+We should clarify in the documentation that H3 works with **topological** hexagons and pentagons, rather than true **geometric** hexagons and pentagons.
+
+See, for example: https://github.com/uber/h3-js/issues/53
+
+
+## Names for H3 Edge Types
+
+Instead of `UnidirectionalEdge`, use the term `DirectedEdge`.
+
+For a future undirected edge mode, use the term `Edge`.
+
+Function name changes:
+
+|                  Current name                 |        Proposed name         |
+|-----------------------------------------------|------------------------------|
+| `getH3UnidirectionalEdge`                     | `getDirectedEdge`            |
+| `h3UnidirectionalEdgeIsValid`                 | `isValidDirectedEdge`        |
+| `getOriginH3IndexFromUnidirectionalEdge`      | `getDirectedEdgeOrigin`      |
+| `getDestinationH3IndexFromUnidirectionalEdge` | `getDirectedEdgeDestination` |
+| `getH3IndexesFromUnidirectionalEdge`          | `getDirectedEdgeCells`       |
+| `getH3UnidirectionalEdgesFromHexagon`         | `getDirectedEdgesFromCell`   |
+| `getH3UnidirectionalEdgeBoundary`             | `getDirectedEdgeBoundary`    |
+
+
+## Function Names
+
+A few more function name change proposals:
+
+|          Current name         |     Proposed name     |                  Notes                  |
+|-------------------------------|-----------------------|-----------------------------------------|
+| (doesn't exist)               | `isValidIndex`        |                                         |
+| `h3IsValid`                   | `isValidCell`         |                                         |
+| `h3UnidirectionalEdgeIsValid` | `isValidDirectedEdge` |                                         |
+| `h3IsPentagon`                | `isPentagon`          |                                         |
+| `h3ToParent`                  | `cellToParent`        |                                         |
+| `h3ToChildren`                | `cellToChildren`      |                                         |
+| `h3IndexesAreNeighbors`       | `cellsAreNeighbors`   |                                         |
+| `geoToH3`                     | `geoToCell`           |                                         |
+| `h3ToGeo`                     | `cellToGeo`           |                                         |
+| `kRing`                       | `disk`                | filled-in disk                          |
+| `hexRing`                     | `ring`                | hollow ring                             |
+| `numHexagons`                 | `numCells`            |                                         |
+| `getRes0Indexes`              | `getRes0Cells`        |                                         |
+| `getPentagonIndexes`          | `getPentagons`        |                                         |
+| `h3GetBaseCell`               | `getBaseCellNumber`   |                                         |
```

Comment:
How about this:

```
- `H3Index`:
    - an unsigned 64-bit integer representing **any** H3 object (hexagon, pentagon, directed edge, ...)
    - often represented as a 15-character (or 16-character) hexadecimal string, like `'8928308280fffff'`
    - the full term "H3 index" should be used to avoid confusion with other common uses of "index"; when a "traditional" index is needed, prefer using "number", "pos", or another term to avoid confusion
```